src/csvcubed/models/cube/cube.py

```diff
@@ -96,4 +103,42 @@ def _validate_columns(self) -> List[ValidationError]:
                         self._get_validation_error_for_exception_in_col(column, e)
                     )
 
+        # Check for uri template naming errors
+        safe_column_names = [
+            csvw_column_name_safe(c.uri_safe_identifier) for c in self.columns
+        ]
+        for uri_template, name in self._csv_column_uri_templates_to_names():
+            if name not in (safe_column_names + URI_TEMPLATE_SPECIAL_PROPERTIES):
+                _logger.debug('Unable to find name %s in %s', name, safe_column_names)
+                errors.append(UriTemplateNameError(safe_column_names, uri_template))
+
         return errors
+
+
+    def _csv_column_uri_templates_to_names(self) -> Iterable[Tuple[str, str]]:
+        """
+        Generates tuples of any configured and not None csv column
+        uri templates within the cube, along with the column name
+        specified within it.
+
+        Example:
+        A cube containing just the following csv column uri templates
+
+        "http://example.com/dimensions/{+foo}"
+        "http://example.com/dimensions/{bar}#things"
+
+        yields:
+        - "http://example.com/dimensions/{+foo}": "foo",
+        then
+        - "http://example.com/dimensions/{bar}#things": "bar"
+        """
+        csv_column_uri_templates = [
+            c.csv_column_uri_template for c in self.columns if hasattr(c, "csv_column_uri_template") # pyright: ignore
```

Comment:
yeah I tried that initially, there's a circular import I couldn't resolve. Happy to take a steer on how or an alternatively but I had no luck unpicking it.

Did get me curious though, is this https://github.com/GSS-Cogs/csvcubed/blob/b597f1ee9e978f1fe36cf8ffedade2b31e6fa8ec/src/csvcubed/models/cube/cube.py#L37  actually be a CsVColumn or are we using the parent class to avoid the circular import?

Reply:
The parent class was there since the cube class was potentially going to be used for different types of outputs (i.e. a QbCube or perhaps we could have configured a CmdCube as well?).

Let's have a call on the circular import issue.